_posts/2017-03-16-Rust-1.16.md

```diff
@@ -0,0 +1,317 @@
+[`IpAddr::is_ipv4`]: https://doc.rust-lang.org/std/net/enum.IpAddr.html#method.is_ipv4
+[`IpAddr::is_ipv6`]: https://doc.rust-lang.org/std/net/enum.IpAddr.html#method.is_ipv6
+[`Result::unwrap_or_default`]: https://doc.rust-lang.org/std/result/enum.Result.html#method.unwrap_or_default
+[`SocketAddr::is_ipv4`]: https://doc.rust-lang.org/std/net/enum.SocketAddr.html#method.is_ipv4
+[`SocketAddr::is_ipv6`]: https://doc.rust-lang.org/std/net/enum.SocketAddr.html#method.is_ipv6
+[`String::insert_str`]: https://doc.rust-lang.org/std/string/struct.String.html#method.insert_str
+[`String::split_off`]: https://doc.rust-lang.org/std/string/struct.String.html#method.split_off
+[`Vec::dedup_by_key`]: https://doc.rust-lang.org/std/vec/struct.Vec.html#method.dedup_by_key
+[`Vec::dedup_by`]: https://doc.rust-lang.org/std/vec/struct.Vec.html#method.dedup_by
+[`VecDeque::resize`]:  https://doc.rust-lang.org/std/collections/vec_deque/struct.VecDeque.html#method.resize
+[`VecDeque::truncate`]: https://doc.rust-lang.org/std/collections/vec_deque/struct.VecDeque.html#method.truncate
+[`str::repeat`]: https://doc.rust-lang.org/std/primitive.str.html#method.repeat
+[`str::replacen`]: https://doc.rust-lang.org/std/primitive.str.html#method.replacen
+
+In addition, a number of small improvements to existing functions landed. For example [`writeln!` now has
+a single-argument form], just like `println!` has. This ends up writing only a newline, but is a nice bit
+of symmetry.
+
+[`writeln!` now has a single-argument form]: https://github.com/rust-lang/rust/pull/38469
+
+All structs in the standard library [now implement `Debug`].
+
+[now implement `Debug`]: https://github.com/rust-lang/rust/pull/38006
+
+When slicing a `&str`, [you'll see better errors]. For example, this code:
+
+```rust
+&"abcαβγ"[..4]
+```
+
+Is incorrect. It generates this error:
+
+```text
+thread 'str::test_slice_fail_boundary_1' panicked at 'byte index 4 is not
+a char boundary; it is inside 'α' (bytes 3..5) of `abcαβγ`'
+```
+
+The part after the `;` is new.
+
+[you'll see better errors]: https://github.com/rust-lang/rust/pull/38066
+
+See the [detailed release notes][notes] for more.
+
+#### Cargo features
+
+In addition to `cargo check`, Cargo and crates.io have some new polish added. For example,
+[`cargo build`] and [`cargo doc`] now take a `--all` flag for building and documenting every
+crate in your workspace with one command.
+
+[`cargo build`]: https://github.com/rust-lang/cargo/pull/3511
+[`cargo doc`]: https://github.com/rust-lang/cargo/pull/3515
+
+Cargo now has a [`--version --verbose` flag], mirroring `rustc`.
+
+[`--version --verbose` flag]: https://github.com/rust-lang/cargo/pull/3604
+
+Crates.io now [can show off your TravisCI or AppVeyor badges] on your crate's page.
+
+[can show off your TravisCI or Appveyor badges]: https://github.com/rust-lang/cargo/pull/3546
+
+In addition, both Cargo and crates.io [understand categories]. Unlike keywords, which are free-form,
```

Comment:
Of this section? I'm not sure. It just felt a bit buried.

But reflecting more, I'm not sure the front-end experience is quite where we want it to make a "big splash" here, so maybe a passing mention is better.